### PR TITLE
add generate method to SpeechT5ForTextToSpeech

### PR DESCRIPTION
**Pull request by ylacombe**
# What does this PR do?

This simple PR aims at adding a `generate` method to `SpeechT5ForTextToSpeech`, which does exactly the same than `generate_speech`. 

`generate_speech` was left for backward compatibility.

The goal is to make `SpeechT5ForTextToSpeech` compatible with the [incoming TTS pipeline](https://github.com/huggingface/transformers/pull/24952) which should not implement any special cases for older models. More on the matter in [this comment](https://github.com/huggingface/transformers/pull/24952#pullrequestreview-1556507240).

- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

## Who can review?

@sanchit-gandhi and @sgugger , WDYT?
Thanks for your help!

### CODE REVIEW

**Comment by sgugger** [comment]
cc @gante as well

File: src/transformers/models/speecht5/modeling_speecht5.py [hunk 1]
```diff
@@ -2783,6 +2783,65 @@ def forward(
...
+                [What are input IDs?](../glossary#input-ids)
+            speaker_embeddings (`torch.FloatTensor` of shape `(batch_size, config.speaker_embedding_dim)`, *optional*):
+                Tensor containing the speaker embeddings.
+            threshold (`float`, *optional*, defaults to 0.5):
+                The generated sequence ends when the predicted stop token probability exceeds this value.
+            minlenratio (`float`, *optional*, defaults to 0.0):
+                Used to calculate the minimum required length for the output sequence.
+            maxlenratio (`float`, *optional*, defaults to 20.0):
+                Used to calculate the maximum allowed length for the output sequence.
+            vocoder (`nn.Module`, *optional*, defaults to `None`):
```
Comment (sgugger) [inline src/transformers/models/speecht5/modeling_speecht5.py:2818]:
```suggestion
            vocoder (`nn.Module`, *optional*):
```
We don't indicate the default when it's `None`.

**Review by sgugger — APPROVED** [review]
Thanks for adding!

**Review by sanchit-gandhi — APPROVED** [review]
LGTM, thanks for updating this @ylacombe! As a final refactor step for SpeechT5, we can add a new model class that combines SpeechT5 with the HiFiGAN vocoder (`SpeechT5ForTextToSpeechWithHiFiGAN `) -> then the user can just call `model.generate` and not have to worry about passing a `vocoder` as argument or loading it standalone

Before:
```python
from transformers import SpeechT5ForTextToSpeech, SpeechT5HifiGan

model = SpeechT5ForTextToSpeech.from_pretrained("microsoft/speecht5_tts")
vocoder = SpeechT5HifiGan.from_pretrained("microsoft/speecht5_hifigan")

speech = model.generate_speech(input_ids, vocoder=vocoder)
```


After:
```python
from transformers import SpeechT5ForTextToSpeechWithHiFiGAN

model = SpeechT5ForTextToSpeechWithHiFiGAN("microsoft/speecht5_tts_vocoder")
speech = model.generate(input_ids)
```

**Comment by ylacombe** [comment]
Hi @sanchit-gandhi and @sgugger , thanks for the review! 

I would like to add `SpeechT5ForTextToSpeechWithHiFiGAN` in another PR if that's ok with you, since it requires additional tests, and since the changes made in the current PR are enough to use `SpeechT5ForTextToSpeech` with the incoming TTS pipeline!

I can open an issue to talk about `SpeechT5ForTextToSpeechWithHiFiGAN` in the meantime if you want,

thanks

**Comment by sanchit-gandhi** [comment]
Yep good with me to add in a follow-up PR!

**Pull request merged**